rare/components/tabs/games/game_widgets/__init__.py

```diff
@@ -0,0 +1,216 @@
+        return self.add_widgets(rgame, utils)
+
+    def add_widgets(self, rgame: RareGame, utils) -> Tuple[IconGameWidget, ListGameWidget]:
+        icon_widget = IconGameWidget(rgame, utils)
+        list_widget = ListGameWidget(rgame, utils)
+        return icon_widget, list_widget
+
+    def __find_widget(
+        self, app_name: str
+    ) -> Tuple[Union[IconGameWidget, None], Union[ListGameWidget, None]]:
+        iw = self._icon_container.findChild(IconGameWidget, app_name)
+        lw = self._list_container.findChild(ListGameWidget, app_name)
+        return iw, lw
+
+    @staticmethod
+    def __visibility(widget, filter_name, search_text):
+        if filter_name == "installed":
+            visible = widget.rgame.is_installed
+        elif filter_name == "offline":
+            visible = widget.rgame.is_offline
+        elif filter_name == "32bit":
+            visible = widget.rgame.is_win32
+        elif filter_name == "mac":
+            visible = widget.rgame.is_mac
+        elif filter_name == "installable":
+            visible = not widget.rgame.is_non_asset
+        elif filter_name == "include_ue":
+            visible = True
+        elif filter_name == "all":
+            visible = not widget.rgame.is_unreal
+        else:
+            visible = True
+
+        if (
+            search_text not in widget.game.app_name.lower()
+            and search_text not in widget.game.app_title.lower()
+        ):
+            opacity = 0.25
+        else:
+            opacity = 1
+
+        return visible, opacity
+
+    def filter_list(self, filter_name="all", search_text: str = ""):
+        icon_widgets = self._icon_container.findChildren(IconGameWidget)
+        list_widgets = self._list_container.findChildren(ListGameWidget)
+        for iw in icon_widgets:
+            visibility, opacity = self.__visibility(iw, filter_name, search_text)
+            iw.setOpacity(opacity)
+            iw.setVisible(visibility)
+        for lw in list_widgets:
+            visibility, opacity = self.__visibility(lw, filter_name, search_text)
+            lw.setOpacity(opacity)
+            lw.setVisible(visibility)
+        self.sort_list(search_text)
+
+    @pyqtSlot()
+    def sort_list(self, sort_by: str = ""):
+        # lk: this is the existing sorting implemenation
+        # lk: it sorts by installed then by title
+        if sort_by:
+            self._icon_container.layout().sort(
+                lambda x: (sort_by not in x.widget().rgame.app_title.lower(),)
+            )
+        else:
+            self._icon_container.layout().sort(
+                lambda x: (
+                    not x.widget().rgame.is_installed,
+                    x.widget().rgame.is_non_asset,
+                    x.widget().rgame.app_title,
+                )
+            )
+        list_widgets = self._list_container.findChildren(ListGameWidget)
+        if sort_by:
+            list_widgets.sort(key=lambda x: (sort_by not in x.game.app_title.lower(),))
```

Comment:
Hm, why are you enforcing the tuple here?

Reply:
Possibly a remnant of trying multiple conditions at some point.